.github/workflows/hls.yml

```diff
@@ -8,53 +8,35 @@ on:
 
 jobs:
 
-  # --------
-  # We build CIRCT in a separate job.
-  # This makes it more likely that when rerunning failed jobs it's
-  # not necessary to rebuild CIRCT.
-  # --------
-
-  circt:
-    runs-on: ubuntu-22.04
-    steps:
-    - uses: actions/checkout@v3
-      with:
-        fetch-depth: '0'
-    - name: "Build CIRCT"
-      uses: ./.github/actions/BuildCirct
-
   hls-test-suite:
     runs-on: ubuntu-22.04
-    needs: circt
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: '0'
 
-    - name: "Install LLVM, Clang, MLIR, and Ninja"
-      uses: ./.github/actions/InstallLlvmDependencies
-
-    - name: "Build CIRCT"
-      uses: ./.github/actions/BuildCirct
-
     # --------
-    # Compile MLIR version of jlm
+    # Compile jlm with the HLS backend enabled
     # --------
 
-    - name: Configure jlm
-      run: ./configure.sh --circt-path $(pwd)/jlm-eval-suite/circt/local CXX=clang++-16
-    - name: Compile jlm
+    - name: "Build CIRCT"
+      uses: ./.github/actions/BuildCirct
+    - name: "Configure jlm"
+      run: ./configure.sh --enable-hls ${{ github.workspace }}/build-circt/circt CXX=clang++-15
```

Comment:
`CXX=clang++-16` and not clang++-15

Reply:
This only provides the compiler for building jlm, not the library dependencies. In other words, you build jlm that depends on LLVM 16 with the clang15, which is perfectly fine to do, but not what we really wanted.